src/librustc_mir/util/liveness.rs

```diff
@@ -33,32 +33,75 @@
 //! generator yield points, all pre-existing references are invalidated, so this
 //! doesn't matter).
 
-use rustc::mir::*;
+use rustc::mir::visit::MirVisitable;
 use rustc::mir::visit::{PlaceContext, Visitor};
-use rustc_data_structures::indexed_vec::{Idx, IndexVec};
+use rustc::mir::Local;
+use rustc::mir::*;
+use rustc::ty::{item_path, TyCtxt, TypeFoldable};
 use rustc_data_structures::indexed_set::IdxSetBuf;
+use rustc_data_structures::indexed_vec::{Idx, IndexVec};
 use rustc_data_structures::work_queue::WorkQueue;
-use util::pretty::{dump_enabled, write_basic_block, write_mir_intro};
-use rustc::ty::item_path;
-use rustc::mir::visit::MirVisitable;
-use std::path::{Path, PathBuf};
 use std::fs;
-use rustc::ty::TyCtxt;
 use std::io::{self, Write};
+use std::path::{Path, PathBuf};
 use transform::MirSource;
+use util::pretty::{dump_enabled, write_basic_block, write_mir_intro};
 
-pub type LocalSet = IdxSetBuf<Local>;
+pub type LocalSet<V> = IdxSetBuf<V>;
```

Comment:
oh, let's rename this to `LiveVarSet<V>` -- it'd be nice to distinguish *locals* (all variables) from *live variables* (the set of variables we are computing liveness on). Maybe not the best name, but it works (and fits with e.g. `LiveVariableMap` etc)